src/nvc.rs

```diff
@@ -155,10 +257,19 @@ impl Nvc {
         self.psw_interrupt_mask_level = ((value as usize) >> 16) & 0x0f;
     }
 
+    pub fn fetch_instruction(&mut self, interconnect: &mut Interconnect, addr: u32) -> (u16, bool) {
+        let halfword = interconnect.read_halfword(addr);
+        if !self.chcw_cache_enabled || !self.cpu_cache.lookup(addr) {
+            return (halfword, false);
+        }
+        return (halfword, true);
+    }
+
     pub fn step(&mut self, interconnect: &mut Interconnect) -> (usize, bool) {
         let original_pc = self.reg_pc;
 
-        let first_halfword = interconnect.read_halfword(original_pc);
+        let (first_halfword, first_cached) = self.fetch_instruction(interconnect, original_pc);
```

Comment:
I think I'd vote for stripping this part of the return value from the interface as it's not currently used, but since we know that's coming, not a big deal. However, we can remove the extra `_` bindings by just binding with `_` directly, as in:

```rust
let (first_halfword, _) = self.fetch_instruction(interconnect, original_pc);
```